rust/src/arrow/linalg.rs

```diff
@@ -130,13 +129,46 @@ impl MatrixView {
         }
     }
 
-    pub fn row(&self, i: usize) -> Option<Float32Array> {
+    /// Returns a row at index `i`. Returns `None` if the index is out of bound.
+    ///
+    /// # Panics if the matrix is transposed.
```

Comment:
Why Panic instead of Result?

Reply:
this is assert. which should not happen , and there is no code path to recover from such error?